### PR TITLE
Refactor to view binding

### PR DESCRIPTION
**Pull request by etman55**

### CODE REVIEW

**Comment by jreij** [comment]
@etman55 what about these 3 classes:
- `InstallmentListAdapter`
- `PaymentMethodAdapter`
- `CountryViewHolder`

Can we use view binding inside them as well?

File: example-app/src/main/java/com/adyen/checkout/example/ui/configuration/ConfigurationActivity.kt [hunk 1]
```diff
@@ -12,12 +12,14 @@ import android.os.Bundle
...
 import androidx.preference.PreferenceFragmentCompat
 import com.adyen.checkout.example.R
+import com.adyen.checkout.example.databinding.ActivitySetttingsBinding
 
 class ConfigurationActivity : AppCompatActivity() {
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
-        setContentView(R.layout.activity_setttings)
+        val binding = ActivitySetttingsBinding.inflate(layoutInflater)
```
Comment (ozgur00) [inline example-app/src/main/java/com/adyen/checkout/example/ui/configuration/ConfigurationActivity.kt:21]: can you fix the typo in the name of the layout?

File: issuer-list/src/main/java/com/adyen/checkout/issuerlist/IssuerListRecyclerAdapter.kt [hunk 2]
```diff
@@ -43,22 +41,22 @@ internal class IssuerListRecyclerAdapter(
...
+    ) : RecyclerView.ViewHolder(binding.root) {
 
         init {
-            logoImage.isVisible = !hideIssuerLogo
+            binding.imageViewLogo.isVisible = !hideIssuerLogo
         }
 
         fun bind(paymentMethod: String, issuerModel: IssuerModel, hideIssuerLogo: Boolean, imageLoader: ImageLoader) {
-            text.text = issuerModel.name
+            binding.textViewText.text = issuerModel.name
```
Comment (ozgur00) [inline issuer-list/src/main/java/com/adyen/checkout/issuerlist/IssuerListRecyclerAdapter.kt:54]: `textViewText` doesn't look that great. maybe change it to `textViewTitle`?

File: drop-in/src/main/java/com/adyen/checkout/dropin/ui/paymentmethods/PaymentMethodAdapter.kt [hunk 3]
```diff
@@ -81,172 +91,185 @@ class PaymentMethodAdapter @JvmOverloads constructor(
...
-        val storedPaymentMethod = getStoredPaymentMethodAt(position)
-
-        when (storedPaymentMethod) {
-            is StoredCardModel -> bindStoredCard(holder, storedPaymentMethod)
-            is GenericStoredModel -> bindGenericStored(holder, storedPaymentMethod)
-        }
-
-        holder.underlayButton.setOnClickListener {
-            showRemoveStoredPaymentDialog(holder.itemView, storedPaymentMethod)
+    class StoredPaymentMethodVH(
```
Comment (OscarSpruit) [inline drop-in/src/main/java/com/adyen/checkout/dropin/ui/paymentmethods/PaymentMethodAdapter.kt:113]: ViewHolder classes should be private

File: drop-in/src/main/java/com/adyen/checkout/dropin/ui/paymentmethods/PaymentMethodAdapter.kt [hunk 4]
```diff
@@ -81,172 +91,185 @@ class PaymentMethodAdapter @JvmOverloads constructor(
...
+    class NoteVH(private val binding: PaymentMethodsListNoteBinding) : BaseViewHolder(binding) {
+        fun bind(model: PaymentMethodNote) = with(binding) {
+            paymentMethodNote.text = model.note
+        }
     }
 
-    companion object {
-        private val TAG = LogUtil.getTag()
-    }
+    open class BaseViewHolder(binding: ViewBinding) : RecyclerView.ViewHolder(binding.root)
```
Comment (OscarSpruit) [inline drop-in/src/main/java/com/adyen/checkout/dropin/ui/paymentmethods/PaymentMethodAdapter.kt:272]: This class is useless, might as well replace it with just `RecyclerView.ViewHolder`

**Comment by OscarSpruit** [comment]
We should take a look at all RecyclerView adapters sometime, I think we can improve them a lot

File: drop-in/src/main/java/com/adyen/checkout/dropin/ui/paymentmethods/PaymentMethodAdapter.kt (hunk 4 above)
Comment (etman55) [reply drop-in/src/main/java/com/adyen/checkout/dropin/ui/paymentmethods/PaymentMethodAdapter.kt:272]: agree

**Review by OscarSpruit — APPROVED** [review]

**Pull request merged**